kaggle-classification/keras_trainer/rnn.py

```diff
@@ -24,15 +24,23 @@ def __init__(self, embeddings_matrix, hparams):
     self.hparams = hparams
 
   def get_model(self):
-    I = Input(shape=(self.hparams.sequence_length,), dtype='float32')
+    sequence_length = self.hparams.sequence_length
+
+    I = Input(shape=(sequence_length,), dtype='float32')
     E = Embedding(
         self.hparams.vocab_size,
         self.hparams.embedding_dim,
         weights=[self.embeddings_matrix],
         input_length=self.hparams.sequence_length,
         trainable=self.hparams.train_embedding)(
             I)
-    X = GRU(128, return_sequences=False)(E)
+    H = Bidirectional(GRU(128, return_sequences=True))(E)
+    A = TimeDistributed(Dense(3), input_shape=(sequence_length, 256))(H)
+    A = Flatten()(A)
+    A = Dense(sequence_length, activation='softmax')(A)
+    A = Reshape((250, 1))(A)
+    X = Multiply()([H, A])
+    X = Flatten()(X)
```

Comment:
I'm not sure this is exactly right. Don't you want to sum the weight multiplied hidden units, not flatten them? Otherwise, you're just multiplying by a constant and passing the hidden units along which doesn't feel like it does enough.

Reply:
You're right. Replaced with average pooling... I can't seem to find a way to sum over a dimension. Will try implementing a custom layer later.